bundles/org.eclipse.mita.program/src/org/eclipse/mita/program/generator/StatementGenerator.xtend

```diff
@@ -301,7 +301,7 @@ class StatementGenerator {
 
 					.data.«altAccessor» = («dataType») {
 						«FOR i_arg: stmt.arguments.indexed SEPARATOR(',\n')»
-						«accessor(feature, i_arg.value.parameter, ".",  " = ").apply(i_arg.key)»«i_arg.value.value.code.noTerminator»
+						«IF i_arg.value.parameter !== null»«accessor(feature, i_arg.value.parameter, ".",  " = ").apply(i_arg.key)»«ENDIF»«i_arg.value.value.code.noTerminator»
```

Comment:
This branch generates the following:
```typescript
struct asdf {
	var x: int32;
}
alt foo { bar: int32 | baz: {x: int32} | ball: asdf }
let x = foo.bar(1);
let y = foo.baz(1);
let z = foo.ball(1);
```

```c
foo x = {
	.tag = foo_bar_e, 
	
	.data.bar = (int32_t) {
		1
	}
};
foo y = {
	.tag = foo_baz_e, 
	
	.data.baz = (baz_t) {
		1
	}
};
foo z = {
	.tag = foo_ball_e, 
	
	.data.ball = (asdf) {
		1
	}
};
```

but it should generate:

```c
foo x = {
	.tag = foo_bar_e, 
	
	.data.bar = (int32_t) {
		1
	}
};
foo y = {
	.tag = foo_baz_e, 
	
	.data.baz = (baz_t) {
		.x = 1
	}
};
foo z = {
	.tag = foo_ball_e, 
	
	.data.ball = (asdf) {
		.x = 1
	}
};
```

Can you please modify your fix to accommodate these [designated inits](https://gcc.gnu.org/onlinedocs/gcc-4.4.1/gcc/Designated-Inits.html#Designated-Inits) where they are available? IMHO this is better/more readable code.

It's probably enough to check if the type that is being initialized is primitive, i.e. if `feature.realType instanceof PrimitiveType`